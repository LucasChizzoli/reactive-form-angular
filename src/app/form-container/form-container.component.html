<div class="form-container-wrapper">
  <h3 class="form-container-title">{{formTitle}}</h3>
  <ng-content></ng-content>
</div>
