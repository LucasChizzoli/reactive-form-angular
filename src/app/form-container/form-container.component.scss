.form-container-wrapper {
  min-width: 400px;

  .form-container-title {
    font-size: 2rem;
    text-align: center;
    margin: 0;
  }
}
