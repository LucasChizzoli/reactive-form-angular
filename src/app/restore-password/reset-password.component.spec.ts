import { ComponentFixture, TestBed } from '@angular/core/testing';
import { ResetPasswordComponent } from './reset-password.component';
import { FormContainerComponent } from '../form-container/form-container.component';
import { FormManagerComponent } from '../form-manager.component';
import { FormsModule, ReactiveFormsModule, Validators } from '@angular/forms';

describe('RestorePasswordComponent', () => {
  let component: ResetPasswordComponent;
  let fixture: ComponentFixture<ResetPasswordComponent>;

  beforeEach(async () => {
    await TestBed.configureTestingModule({
      imports: [
        FormsModule,
        ReactiveFormsModule
      ],
      declarations: [ ResetPasswordComponent, FormContainerComponent, FormManagerComponent ]
    })
    .compileComponents();
  });

  beforeEach(() => {
    fixture = TestBed.createComponent(ResetPasswordComponent);
    component = fixture.componentInstance;
    fixture.detectChanges();
  });

  it('should create', () => {
    expect(component).toBeTruthy();
  });

  test('should have email', () => {
    component.fields = [
      {
        name: 'email',
        validators: [Validators.email, Validators.required]
      },
    ];
    fixture.detectChanges();
    expect(component).toMatchSnapshot();
  });
});
