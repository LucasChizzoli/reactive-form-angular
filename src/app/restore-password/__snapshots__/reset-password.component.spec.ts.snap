// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`RestorePasswordComponent should have email 1`] = `
ResetPasswordComponent {
  "FIELDS": Array [
    Object {
      "name": "email",
      "validators": Array [
        [Function],
        [Function],
      ],
    },
  ],
  "fields": Array [
    Object {
      "name": "email",
      "validators": Array [
        [Function],
        [Function],
      ],
    },
  ],
  "form": FormGroup {
    "_composedAsyncValidatorFn": null,
    "_composedValidatorFn": null,
    "_hasOwnPendingAsyncValidator": false,
    "_onCollectionChange": [Function],
    "_onDisabledChange": Array [],
    "_parent": null,
    "_rawAsyncValidators": null,
    "_rawValidators": null,
    "controls": Object {
      "email": FormControl {
        "_composedAsyncValidatorFn": null,
        "_composedValidatorFn": [Function],
        "_hasOwnPendingAsyncValidator": false,
        "_onChange": Array [
          [Function],
          [Function],
        ],
        "_onCollectionChange": [Function],
        "_onDisabledChange": Array [
          [Function],
          [Function],
        ],
        "_parent": [Circular],
        "_pendingValue": "",
        "_rawAsyncValidators": null,
        "_rawValidators": Array [
          [Function],
          [Function],
        ],
        "errors": Object {
          "required": true,
        },
        "pristine": true,
        "status": "INVALID",
        "statusChanges": EventEmitter_ {
          "__isAsync": false,
          "_isScalar": false,
          "closed": false,
          "hasError": false,
          "isStopped": false,
          "observers": Array [],
          "thrownError": null,
        },
        "touched": false,
        "value": "",
        "valueChanges": EventEmitter_ {
          "__isAsync": false,
          "_isScalar": false,
          "closed": false,
          "hasError": false,
          "isStopped": false,
          "observers": Array [],
          "thrownError": null,
        },
      },
    },
    "errors": null,
    "pristine": true,
    "status": "INVALID",
    "statusChanges": EventEmitter_ {
      "__isAsync": false,
      "_isScalar": false,
      "closed": false,
      "hasError": false,
      "isStopped": false,
      "observers": Array [],
      "thrownError": null,
    },
    "touched": false,
    "value": Object {
      "email": "",
    },
    "valueChanges": EventEmitter_ {
      "__isAsync": false,
      "_isScalar": false,
      "closed": false,
      "hasError": false,
      "isStopped": false,
      "observers": Array [],
      "thrownError": null,
    },
  },
  "getErrorMessage": [Function],
  "getType": [Function],
  "submitted": EventEmitter_ {
    "__isAsync": false,
    "_isScalar": false,
    "closed": false,
    "hasError": false,
    "isStopped": false,
    "observers": Array [],
    "thrownError": null,
  },
}
`;
